plugins/wasmedge_stablediffusion/sd_func.cpp

```diff
@@ -374,12 +438,53 @@ Expect<uint32_t> SDImageToImage::body(
           "[WasmEdge-StableDiffusion] Load image from input image failed."sv);
       return static_cast<uint32_t>(ErrNo::InvalidArgument);
     }
+    if (Channel < 3) {
+      spdlog::error(
+          "[WasmEdge-StableDiffusion] The number of channels for the input image must be >= 3."sv);
+      free(InputImageBuffer);
+      return static_cast<uint32_t>(ErrNo::InvalidArgument);
+    }
+    if (ImageWidth <= 0) {
+      spdlog::error(
+          "[WasmEdge-StableDiffusion] The width of image must be greater than 0."sv);
+      free(InputImageBuffer);
+      return static_cast<uint32_t>(ErrNo::InvalidArgument);
+    }
+    if (ImageHeight <= 0) {
+      spdlog::error(
+          "[WasmEdge-StableDiffusion] The height of image must be greater than 0."sv);
+      free(InputImageBuffer);
+      return static_cast<uint32_t>(ErrNo::InvalidArgument);
+    }
+
+    // Resize input image ...
+    if (Height != static_cast<uint32_t>(ImageHeight) ||
+        Width != static_cast<uint32_t>(ImageWidth)) {
+      uint8_t ResizedHeight = Height;
+      uint8_t ResizedWidth = Width;
```

Comment:
`Height` and `Width` are `uint32_t` types. Why do we need to reduce the type from `uint32_t` to `uint8_t`?

Reply:
It's pretty non sense that the type of `InputImageBuffer` will affect the `ResizedHeight` and `ResizedWidth`. The maximum value of `uint8_t` is 255. You truncated the `Height` and `Width` from `uint32_t` into `uint8_t` so the `ResizedImageBuffer` will be extreme smaller than the original image. I don't think this is expected.

Especially, the [`stbir_resize` function](https://github.com/leejet/stable-diffusion.cpp/blob/master/thirdparty/stb_image_resize.h#L341-L347) supports `int` as the input for both `Height` and `Width`.

Reply:
I'm sorry for that mistake, and I will change it to `int`.